### DISCUSSION TITLE
Gleam stdlib example not working : reference : https://hexdocs.pm/gleam_stdlib/index.html

### DISCUSSION THREAD

**Discussion by functonal** [gleam-lang/gleam#1483] [Questions & support]
As usual :
gleam new gleam_stdlib_project
cd gleam_stdlib_project
<The generated source code was changed >
```
import gleam/string
import gleam/list.{contains}

fn usage() {
  string.append("str", "ing")
}

fn more_usage() {
  [1, 2, 3]
  |> contains(any: 2)
}
```
No pub fn main() here..

Will it compile and test : No
The results are : 
A.
```
➜  gleam_stdlib_project gleam test
  Compiling gleam_stdlib_project

warning: Unused private function
  ┌─ ./src/gleam_stdlib_project.gleam:4:1
  │
4 │ fn usage() {
  │ ^^^^^^^^^^ This private function is never used.

Hint: You can safely remove it.

warning: Unused private function
  ┌─ ./src/gleam_stdlib_project.gleam:7:1
  │
7 │ fn more_usage() {
  │ ^^^^^^^^^^^^^^^ This private function is never used.

Hint: You can safely remove it.
Compiled in 1.30s
Running gleam_stdlib_project_test.main
.
Finished in 0.012 seconds
1 tests, 0 failures
```
B.
```
gleam_stdlib_project gleam run
  Compiling gleam_stdlib_project

warning: Unused private function
  ┌─ ./src/gleam_stdlib_project.gleam:7:1
  │
7 │ fn more_usage() {
  │ ^^^^^^^^^^^^^^^ This private function is never used.

Hint: You can safely remove it.

warning: Unused private function
  ┌─ ./src/gleam_stdlib_project.gleam:4:1
  │
4 │ fn usage() {
  │ ^^^^^^^^^^ This private function is never used.

Hint: You can safely remove it.
Compiled in 1.07s
Running gleam_stdlib_project.main
exception error: undefined function gleam_stdlib_project:main/0%
```
So no main() no invoke ,will not compile and can not run test.

The example is incomplete....given on this link  https://hexdocs.pm/gleam_stdlib/index.html
How to rectify this : I want to use both the private functions usage() and more_usage()
and moreover want to invoke them from Erlang shell via gleam shell..

Could you shed some light on it...

Thanks

**Comment by functonal** [comment]
I found the answer  to this : Gleam stdlib example not working : reference : https://hexdocs.pm/gleam_stdlib/index.html
#1483

The answer is very simple : I have to use 
using keyword pub (from my little bit of Rust Knowledge)

So the code from  https://hexdocs.pm/gleam_stdlib/index.html changed to >>
import gleam/string
import gleam/list.{contains}

pub fn usage() {
string.append("str", "ing")
}
pub  fn more_usage() {
[1, 2, 3]
|> contains(any: 2)

}

and the results are : 
➜  gleam_stdlib_project gleam test
  Compiling gleam_stdlib_project
  Compiled in 1.28s
  Running gleam_stdlib_project_test.main
.
Finished in 0.010 seconds
1 tests, 0 failures
➜  gleam_stdlib_project gleam run
  Compiling gleam_stdlib_project
  Compiled in 0.74s
  Running gleam_stdlib_project.main
  exception error: undefined function gleam_stdlib_project:main/0. //  The Erlang backend 
and the erlang shell from gleam: 
1> gleam_stdlib_project:usage().
<<115,116,114,105,110,103>>
2> gleam_stdlib_project:more_usage().
true


//Fine but no main()...

My question is How to make it completely correct program and how to use the main() here ?
Please shed some light on it.

Thanks

**Reply by lpil** [reply]
That's right, you'll need to use the keyword `pub` to make things public. These docs may help: https://gleam.run/book/tour/functions.html

> My question is How to make it completely correct program and how to use the main() here ?

I'm not sure what you mean, it would depend what your program is intended to do. The `main` function is called first, so any functionality your program should have will be written there.
